app/src/test/kotlin/compiler/e2e/common/E2eAsserter.kt

```diff
@@ -0,0 +1,18 @@
+package compiler.e2e
+
+import compiler.Compiler
+import compiler.common.diagnostics.CompilerDiagnostics
+import compiler.common.diagnostics.Diagnostic
+import kotlin.test.assertContentEquals
+
+object E2eAsserter {
+    fun assertProgramGeneratesDiagnostics(program: String, diagnostics: List<Diagnostic>) {
+        val actualDiagnostics = CompilerDiagnostics()
+        val compiler = Compiler(actualDiagnostics)
+        compiler.process(program.trimIndent().reader())
+        assertContentEquals(diagnostics.asSequence(), actualDiagnostics.diagnostics)
+    }
+    fun assertProgramCorrect(program: String) {
+        assertProgramGeneratesDiagnostics(program, listOf())
```

Comment:
Technically, a program could be correct and still generate non-error diagnostics. Maybe we could change the name or only check for errors here. It's probably not that important as we're probably not gonna generate many warnings, certainly not in current tests.

Reply:
I think that there aren't enough tests for now to pick a good naming convention :thinking:. I assume we'll be adding more of those tests this week, so let's postpone the choice of the naming convention until then.